packages/block-editor/src/store/reducer.js

```diff
@@ -1194,6 +1195,26 @@ export function isTyping( state = false, action ) {
 	return state;
 }
 
+/**
+ * Reducer that returns a new uuid every time a block is added, removed or inserted.
+ *
+ * @param {string} state  uuid string
+ * @param {Object} action Dispated action
+ *
+ * @return {string} Updated state
+ */
+export function __unstableBlocksChangedUUID( state = uuid(), action ) {
+	switch ( action.type ) {
+		case 'REPLACE_INNER_BLOCKS':
+		case 'INSERT_BLOCKS':
+		case 'REMOVE_BLOCKS':
+		case 'REPLACE_BLOCKS':
+		case 'RESET_BLOCKS':
+			return uuid();
```

Comment:
Note that we don't necessarily need to use a uuid, it just seemed convenient. I'm open to whatever naming changes folks prefer too.